### PR TITLE
Bump to 0.6.0

### PR DESCRIPTION
**Pull request by mkitti**
Should we be bumping to 0.6.0 or 0.6.0.dev?

cc: @MilesCranmer

### CODE REVIEW

**Comment by MilesCranmer** [comment]
Not sure. I don’t think 0.5.8 was actually released yet though.

**Comment by mkitti** [comment]
Did you want to release a 0.5.8 from before your pull request?

**Comment by MilesCranmer** [comment]
I think that’s a good idea. What’s the mechanism for doing a release?

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -57,7 +57,7 @@ jobs:
     runs-on: ubuntu-20.04
     strategy:
       matrix:  # using `matrix` to define a constant
-        package: ['julia==0.5.8.dev']
+        package: ['julia==0.6.0.dev']
```
Comment (MilesCranmer) [inline .github/workflows/publish.yml:60]:
```suggestion
        package: ['julia==0.6.0']
```

(So we can create a new release)

**Comment by MilesCranmer** [comment]
I think we're ready to move to 0.6.0 (or at least, release 0.5.8), so packages can use the fixed `optimize={}, threads={}` to launch Julia. Want to create a release?

File: .github/workflows/publish.yml (hunk 1 above)
Comment (MilesCranmer) [reply .github/workflows/publish.yml:60]: Unless we release 0.5.8 on the current master branch; then we could leave this as 0.6.0.dev.

Comment (mkitti) [reply .github/workflows/publish.yml:60]: For 0.5.8, I think we should create a branch before we dropped Python 2.7 support and proceed that way.

**Comment by mkitti** [comment]
I will look at the release procedure this evening. The first target is Pypi, then conda-forge.

**Comment by mkitti** [comment]
I'm going to work on this tomorrow.

File: ci/test-upload/tox.ini [hunk 2]
```diff
@@ -16,7 +16,7 @@ deps =
 
 commands =
     shell-retry --backoff=2 --interval-max=20 --retry-count=30 --verbose -- \
-    pip install --index-url https://test.pypi.org/simple/ julia==0.5.8.dev
+    pip install --index-url https://test.pypi.org/simple/ julia==0.6.0.dev
```
Comment (mkitti) [inline ci/test-upload/tox.ini:19]:
```suggestion
    pip install --index-url https://test.pypi.org/simple/ julia==0.6.0
```

File: docs/source/conf.py [hunk 3]
```diff
@@ -25,9 +25,9 @@
 author = 'The Julia and IPython development teams'
 
 # The short X.Y version
-version = '0.5.8'
+version = '0.6.0'
 # The full version, including alpha/beta/rc tags
-release = '0.5.8.dev'
+release = '0.6.0.dev'
```
Comment (mkitti) [inline docs/source/conf.py:30]:
```suggestion
release = '0.6.0'
```

File: src/julia/release.py [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # This file is executed via setup.py and imported via __init__.py
 
-__version__ = "0.5.8.dev"
+__version__ = "0.6.0.dev"
```
Comment (mkitti) [inline src/julia/release.py:3]:
```suggestion
__version__ = "0.6.0"
```

**Review by mkitti — COMMENTED** [review]
Changing to 0.6.0 for release

**Pull request merged**